packages/cmr-client/src/EarthdataToken.ts

```diff
@@ -0,0 +1,124 @@
+// @ts-nocheck
+import get from 'lodash/get';
+import got from 'got';
+import Logger from '@cumulus/logger';
+
+import { EarthdataGetTokenResponse, EarthdataPostTokenResponse } from './types';
+const log = new Logger({ sender: 'cmr-client' });
+
+const logDetails: { [key: string]: string } = {
+  file: 'cmr-client/CMR.js',
+};
+
+function formatData(
+  username: string,
+  password: string
+) {
+  const credentials = username + ':' + password;
+  const buff = Buffer.from(credentials).toString('base64');
+  let cmrenv = '';
+  if (process.env.CMR_ENVIRONMENT === 'PROD' || process.env.CMR_ENVIRONMENT === 'OPS') {
+    cmrenv = '';
+  } else if (process.env.CMR_ENVIRONMENT === 'UAT' || process.env.CMR_ENVIRONMENT === 'SIT') {
+    cmrenv = process.env.CMR_ENVIRONMENT + '.';
+  } else {
+    throw new TypeError(`Invalid CMR environment: ${process.env.CMR_ENVIRONMENT}`);
+  }
+  const returnarray: Array<string> = [buff, cmrenv];
+  return returnarray;
+}
+
```

Comment:
1. You can get encoded credential with this:
packages/integration-tests/api/EarthdataLogin.js
const auth = base64.encode(`${username}:${password}`);
2. Instead of checking env, we can pass in `edlEnv` variable (possible values should be PROD, OPS etc.), this way, the methods are more generic.
3. Create a method to return EDL url with env, default to SIT env.
4. It's better to create a EarthdataToken class with attribute username, password, edlEnv.  
packages/oauth-client/src can give you some ideas how to refactor the code.

Reply:
2. Instead of checking env, we can pass in `edlEnv` variable (possible values should be PROD, OPS etc.), this way, the methods are more generic.

Reply:
3. Create a method to return EDL url given edl env, default to SIT env.

Reply:
4. It's better to create a EarthdataLogin class with attribute username, password, edlEnv.  
packages/oauth-client/src can give you some ideas how to refactor the code.